listenbrainz/tests/integration/test_pinned_recording_api.py

```diff
@@ -0,0 +1,470 @@
+
+    def test_get_pins_for_user_following(self):
+        """Test that valid response is received with 200 code"""
+        # user follows followed_user_1 and followed_user_2
+        db_user_relationship.insert(self.user["id"], self.followed_user_1["id"], "follow")
+        db_user_relationship.insert(self.user["id"], self.followed_user_2["id"], "follow")
+
+        pin1 = self.pin_single_sample(self.followed_user_1["id"], 0)  # pin recording for followed_user_1
+        pin2 = self.pin_single_sample(self.followed_user_2["id"], 1)  # pin recording for followed_user_2
+
+        response = self.client.get(
+            url_for("pinned_rec_api_bp_v1.get_pins_for_user_following", user_name=self.user["musicbrainz_id"])
+        )
+        self.assert200(response)
+        data = json.loads(response.data)
+
+        print(json.dumps(data, indent=2))
+        pins = data["pinned_recordings"]
+
+        self.assertEqual(data["offset"], 0)
+        self.assertEqual(data["count"], 2)
+        self.assertEqual(data["user_name"], self.user["musicbrainz_id"])
+
+        # check that data is sorted in descending order of created date
+        self.assertEqual(pins[0]["recording_mbid"], pin2.recording_mbid)  # pin2 was pinned most recently
+        self.assertEqual(pins[0]["blurb_content"], pin2.blurb_content)
+        self.assertEqual(pins[0]["user_name"], self.followed_user_2["musicbrainz_id"])
+
+        self.assertEqual(pins[1]["recording_mbid"], pin1.recording_mbid)  # pin2 was pinned before pin1
+        self.assertEqual(pins[1]["blurb_content"], pin1.blurb_content)
+        self.assertEqual(pins[1]["user_name"], self.followed_user_1["musicbrainz_id"])
+
+    def test_get_pins_for_user_following_invalid_username(self):
+        """Tests that endpoint returns 404 when no user with given user_name is found"""
+        response = self.client.get(
+            url_for("pinned_rec_api_bp_v1.get_pins_for_user_following", user_name=" -- invalid username -- ")
+        )
+        self.assert404(response)
+        self.assertEqual(response.json["code"], 404)
+
+    def test_get_pins_for_user_count_param(self):
+        """Tests that valid response is received honoring count parameter"""
+        # user follows followed_user_1 and followed_user_2
+        db_user_relationship.insert(self.user["id"], self.followed_user_1["id"], "follow")
+        db_user_relationship.insert(self.user["id"], self.followed_user_2["id"], "follow")
+
+        self.pin_single_sample(self.followed_user_1["id"], 0)  # pin recording for followed_user_1
+        included_pin = self.pin_single_sample(self.followed_user_2["id"], 1)  # pin recording for followed_user_2
+
+        limit = 1
+        response = self.client.get(
+            url_for("pinned_rec_api_bp_v1.get_pins_for_user_following", user_name=self.user["musicbrainz_id"], count=limit)
+        )
+        data = json.loads(response.data)
+        pins = data["pinned_recordings"]
+        print(json.dumps(data, indent=2))
```

Comment:
'aight, scan the module for any other prints. :)

Reply:
Oops. Done